src/Storage/Mysql/UserRepository.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Bugcache\Storage\Mysql;
+
+use Amp\Mysql\Pool;
+use Amp\Mysql\ResultSet;
+use Amp\Promise;
+use Bugcache\Storage;
+use Generator;
+use function Amp\resolve;
+
+class UserRepository implements Storage\UserRepository {
+    private $mysql;
+
+    public function __construct(Pool $mysql) {
+        $this->mysql = $mysql;
+    }
+
+    public function findByName(string $username): Promise {
+        $sql = "SELECT id, name FROM users WHERE name = ?";
+
+        return resolve($this->fetchUser($sql, [$username]));
+    }
+
+    private function fetchUser(string $sql, array $params = []): Generator {
+        /** @var ResultSet $result */
+        $result = yield $this->mysql->prepare($sql, $params);
```

Comment:
As said:
> But we possibly might want to have our own interface specifying startup() method or similar?

hummm … yeah, prepares are always executed on the same connection they are established. I thought I implemented it in a way so that it chooses a currently unused connection (preferring one with an existing prepare) and eventually re-prepares. But that's something I can fix in amp/mysql.

The idea was that we end up after little warming up with a large pool of connections where all the queries are prepared once and just need to be executed.